spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -152,10 +151,11 @@ public void doWith(Method method) throws IllegalArgumentException {
 						MessageChannel sharedChannel = locateSharedChannel(name);
 						if (sharedChannel == null) {
 							outputHolders.put(name, new ChannelHolder(
-									channelFactory.createChannel(name, method.getReturnType()), true));
+									channelFactory.createBindableChannel(name, (Class<? extends MessageChannel>) method.getReturnType()), true));
 						}
 						else {
-							configureSharedMessageChannel(name, method.getReturnType(), sharedChannel);
+							configureSharedMessageChannel(name, (Class<? extends MessageChannel>)method.getReturnType(),
```

Comment:
same thing here... could move the `method.getReturnType()` up so it's not duplicated, but if left here add a space after the cast parens